src/lib/util.ts

```
@@ -106,3 +105,26 @@ export const terminalStyle = {
+  ['GH7ome3EiwEr7tu9JuTh2dpYWBJK3z69Xm1ZE3MEE6JC', RpcChain.Devnet], // Solana Devnet
+])
+
+
+// A valid Solana or SVM RPC URL (including localnet) should be able to get a response from the getGenesisHash method
+export const chain = async (rpcUrl: string): Promise<RpcChain> => {
+  try {
+    const connection = new Connection(rpcUrl)
+    const genesisHash = await connection.getGenesisHash()
+    return GENESIS_HASH_MAP.get(genesisHash) ?? RpcChain.Localnet
+  } catch (error) {
+    throw new Error('Could not determine a genesis hash from RPC URL: ' + rpcUrl + '\nPlease set a different RPC URL in your config file or use the --rpc-url flag')
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)